### ISSUE TITLE
[Windows] Compiling with openssl v1.1.0c fails due to unresolved external symbols

### ISSUE THREAD

**Issue by Collisionc** [nzbget/nzbget#317]
Using vs2015 + openssl v.1.1.0c on windows 10, all library/include folders are setup properly. 

Error message:
1>TlsSocket.obj : error LNK2019: unresolved external symbol _OPENSSL_init_crypto referenced in function "public: static void __cdecl TlsSocket::Init(void)" (?Init@TlsSocket@@SAXXZ)
1>TlsSocket.obj : error LNK2019: unresolved external symbol _SSL_CTX_set_options referenced in function "public: bool __thiscall TlsSocket::Start(void)" (?Start@TlsSocket@@QAE_NXZ)
1>TlsSocket.obj : error LNK2019: unresolved external symbol _TLS_method referenced in function "public: bool __thiscall TlsSocket::Start(void)" (?Start@TlsSocket@@QAE_NXZ)
1>TlsSocket.obj : error LNK2019: unresolved external symbol _OPENSSL_init_ssl referenced in function "public: static void __cdecl TlsSocket::Init(void)" (?Init@TlsSocket@@SAXXZ)
1>RarReader.obj : error LNK2019: unresolved external symbol _EVP_MD_CTX_new referenced in function "private: bool __thiscall RarVolume::DecryptRar3Prepare(unsigned char const * const)" (?DecryptRar3Prepare@RarVolume@@AAE_NQBE@Z)
1>RarReader.obj : error LNK2019: unresolved external symbol _EVP_MD_CTX_free referenced in function "private: bool __thiscall RarVolume::DecryptRar3Prepare(unsigned char const * const)" (?DecryptRar3Prepare@RarVolume@@AAE_NQBE@Z)

Adding libssl32MT.lib and libcrypto32MT.lib to the list of dependencies required resolves this.

**Comment by Collisionc** [comment]
Looks like this was changed with openssl 1.1.0, so <1.1.0 should still work without changes. 

This means that libcrypto-1.1.dll and libssl-1_1.dll will need to be provided for the .exe to run, if compiled with openssl >= 1.1.0

**Comment by hugbug** [comment]
That may also depend on how openssl was built. Have you built it yourself or use precompiled binaries? From where (if precompiled)?

**Comment by Collisionc** [comment]
Built myself, precompiled binaries work without any issues as they are compiled with openssl 1.0.2h.

This is only an issue if someone tries to follow the instructions to compile at - https://github.com/nzbget/nzbget/wiki/Installation-on-Windows - and try to use openssl >= 1.1.0, as that is the current latest version. 

Compiling with older versions of openssl still works.

**Comment by Collisionc** [comment]
Actually, the version I compiled with openssl 1.1.0 didn't even work. Once I tried to download something it gave errors:
Mon Dec 12 2016 15:31:29	Could not create TLS context: error:140A90A1:lib(20):func(169):reason(161)

So, I went and just compiled it with openssl 1.0.2j instead, and now it works without any issues.

This is mostly for others who try to compile with openssl 1.1.0 and wonder why it will not work. :)

**Comment by hugbug** [comment]
>Built myself, precompiled binaries work without any issues as they are compiled with openssl 1.0.2h.

I meant whether you have compiler openssl yourself.

I'm using OpenSSL binaries from https://slproweb.com/products/Win32OpenSSL.html.

I've made a test with 1.1.0 and it worked without issues, after I replaced ssleay32MT.lib and
libeay32MT.lib with libssl32MT.lib and libcrypto32MT.lib. In fact support for 1.1.0 was already added to NZBGet in #298 (some extra work were required).

According to openssl web site version 1.0.2 is a Long Term Support (LTS) version. I think I better keep continue using it in Windows builds for a while.

I've added a note about libraries to guide [Installation on Windows](https://github.com/nzbget/nzbget/wiki/Installation-on-Windows#compiling).
